include/graphblas/algorithms/amg/multigrid_v_cycle.hpp

```diff
@@ -0,0 +1,327 @@
+
+/*
+ *   Copyright 2022 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ * This file contains the routines for multi-grid solution refinement,
+ *        including the main routine and those for coarsening and refinement
+ *        of the tentative solution.
+ * @file multigrid_v_cycle.hpp
+ * @author Alberto Scolari (alberto.scolari@huawei.com)
+ * @author Denis Jelovina (denis.jelovina@huawei.com)
+ * @date 2022-08-10
+ */
+
+#ifndef _H_GRB_ALGORITHMS_MULTIGRID_V_CYCLE
+#define _H_GRB_ALGORITHMS_MULTIGRID_V_CYCLE
+
+#include <cassert>
+#include <vector>
+#include <graphblas.hpp>
+#include "amg_data.hpp"
+#include <utils/print_vec_mat.hpp>
+
+#define DBG_println( args ) std::cout << args << std::endl;
```

Comment:
Please remove this -- this will leak to the user and libraries should do that preferably never. Better to incur more typing in this file. The copious use of macros is fine in test `cpp` files only, since macros there will never leak to end users